packages/kbn-typed-react-router-config/src/create_router.ts

```diff
@@ -59,6 +59,17 @@ export function createRouter<TRoutes extends Route[]>(routes: TRoutes): Router<T
       optional = args[1];
     }
 
+    if (location.pathname === '' && location.hash.length > 0) {
+      // We just want the search and pathname so the host doesn't matter
+      const resolvedUrl = new URL(location.hash.slice(1), 'http://localhost');
+      location = {
+        ...location,
+        hash: '',
+        pathname: resolvedUrl.pathname,
+        search: resolvedUrl ? resolvedUrl.search : location.search,
+      };
+    }
+
```

Comment:
@dgieselaar I'm not sure if here is the best place to add this. Let me know what you think.

Reply:
Or if this should be a route-specific property, like we were doing before. Since this will replace all hash-based routes.

Reply:
where was this code previously?

Reply:
It was defined on each route as needed. https://github.com/elastic/kibana/blob/7.13/x-pack/plugins/apm/public/components/app/Main/route_config/index.tsx#L173

Reply:
Ok, we can do the same here I think. It should be added to the last route in `x-pack/plugins/apm/public/components/routing/home/index.tsx`.